dask_geopandas/geohash.py

```diff
@@ -0,0 +1,245 @@
+    x = x @ __s1
+    # Interleave x and y bits so that x and y occupy even and odd bit levels
+    x = x[:, 0] | x[:, 1]
+    x = x >> 4
+
+    return x
+
+
+def _encode_base32(encoded_uint64):
+    """
+    Encode quantized coordinates into base32 pairs
+    Encoding starts at the highest bit, consuming 5 bits for each character precision.
+    This means encoding happens 12 times for the 12 character precision or 60 bits.
+
+    Implementation is based on "Geohash in Golang Assembly"
+    blog (https://mmcloughlin.com/posts/geohash-assembly)
+
+    Parameters
+    ----------
+    g_uint64 : array_like
+        coordinate pairs encoded to uint64 values
+
+    Returns
+    ---------
+    array_like of shape (n, 12)
+        with base 32 values of type unasigned integer
+    """
+    # Define 32 bit mask
+    mask = np.uint64(0x1F).flatten()  # equivelant to 32-1
+    # Return array for each character
+    c11 = (encoded_uint64 >> 0) & mask
+    c10 = (encoded_uint64 >> 5) & mask
+    c9 = (encoded_uint64 >> 10) & mask
+    c8 = (encoded_uint64 >> 15) & mask
+    c7 = (encoded_uint64 >> 20) & mask
+    c6 = (encoded_uint64 >> 25) & mask
+    c5 = (encoded_uint64 >> 30) & mask
+    c4 = (encoded_uint64 >> 35) & mask
+    c3 = (encoded_uint64 >> 40) & mask
+    c2 = (encoded_uint64 >> 45) & mask
+    c1 = (encoded_uint64 >> 50) & mask
+    c0 = (encoded_uint64 >> 55) & mask
+
+    # Stack each array vertically
+    return np.uint8(np.column_stack((c0, c1, c2, c3, c4, c5, c6, c7, c8, c9, c10, c11)))
+
+
+def _encode_unicode(encoded_base32, precision, raw=False):
+    """
+    Encode base32 pairs into geohash bytes with an option to return
+    the geohash in unicode format
+
+    Parameters
+    ----------
+    encoded_base32 : array_like
+        coordinate pairs
+    precision : int
+        precision of the Geohash
+    raw : bool
+        to convert S12 bytes to unicode type
+    Returns
+    ---------
+    array_like of shape (n, precision)
+        containing geohash for a given precision
+    """
+
+    # Define replacement values
+    replacement = np.array(
+        [
+            48,
+            49,
+            50,
+            51,
+            52,
+            53,
+            54,
+            55,
+            56,
+            57,
+            98,
+            99,
+            100,
+            101,
+            102,
+            103,
+            104,
+            106,
+            107,
+            109,
+            110,
+            112,
+            113,
+            114,
+            115,
+            116,
+            117,
+            118,
+            119,
+            120,
+            121,
+            122,
+        ],
+        dtype="uint8",
+    )
+
+    encoded_base32 = replacement[encoded_base32]
+
+    if raw is True:
+        encoded_base32 = encoded_base32.view(np.dtype("|S12"))
+        return encoded_base32.flatten().astype(f"U{precision}")
+
+    elif raw is not True:
```

Comment:
```suggestion
    if raw:
        encoded_base32 = encoded_base32.view(np.dtype("|S12"))
        return encoded_base32.flatten().astype(f"U{precision}")
    else:
```

(although `raw is True` is certainly more explicit, it's general practice to exclude the `is True`. See eg https://www.python.org/dev/peps/pep-0008/#programming-recommendations, one of the bullet points "Don't compare boolean values to True or False using ==")

Reply:
thanks, I need to familiarise myself better with PEP 8